### PR TITLE
TODO update / Generic House-Clean

### PR DESCRIPTION
**Pull request by luke-hill**
- Cleared a few tasks off the list
- Added / Removed TODO items
- Cleanup: Removed all traces of yard/redcarpet
- Added version spec primitive
- page and section specs amended - (Cleaned up some code, removed some duplicate specs, shaved a couple of seconds off runtime)

P.S. Ignore the commit name / message, it's just compiled of several commits rebased into one.

@tgaff @TheMetalCode to review (Kinda, there's nothing much here really)

### CODE REVIEW

File: spec/version_spec.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'SitePrism::VERSION' do
+  subject { SitePrism::VERSION }
+
+  it { is_expected.to eq('2.11') }
```
Comment (TheMetalCode) [inline spec/version_spec.rb]: I'm somewhat skeptical of the value of this test. It basically says "Ruby has class constants and they work" which isn't a thing site_prism is really responsible for, plus this is a test that you'll definitely have to update every time you cut a release. Tests can sometimes be valuable just for the sake of documentation, but even on that score if I wanted to know the version I'd check the gemspec or, absent that, whatever file was defining the `VERSION` constant. It's not something I'd go looking for in the rspec tests.

I wouldn't block the PR over this - it's not a _lot_ of extra maintenance added when bumping version. Just not sure it's really necessary is all.

Comment (TheMetalCode) [inline spec/version_spec.rb]: I'm somewhat skeptical of the value of this test. It basically says "Ruby has class constants and they work" which isn't a thing site_prism is really responsible for, plus this is a test that you'll definitely have to update every time you cut a release. Tests can sometimes be valuable just for the sake of documentation, but even on that score if I wanted to know the version I'd check the gemspec or, absent that, whatever file was defining the VERSION constant. It's not something I'd go looking for in the rspec tests.

I wouldn't block the PR over this - it's not a lot of extra maintenance added when bumping version. Just not sure it's really necessary is all.

Comment (tgaff) [reply spec/version_spec.rb]: Perhaps just verify the constant is set.

Comment (luke-hill) [reply spec/version_spec.rb]: I think actually I'd agree with what you've said, but it's there for precisely that reason. Just to add extra compliance to a codebase I'm not yet fully familiar with. It also adds a spec which would stop accidental versions being bumped.

But having said that, you've both come onboard to review it, which I'm greatful for, as it gives more than just my eyes on it. So I'll alter it to `be_truthy` sometime in the next couple of days.

**Pull request merged**